### PR TITLE
updating deployment to use both albs and elbs

### PR DESCRIPTION
**Pull request by bminahan-kc**

### CODE REVIEW

File: broker-deploy/broker-deploy.tf [hunk 1]
```diff
@@ -18,7 +18,8 @@ resource "aws_autoscaling_group" "api-asg" {
   health_check_type         = "ELB"
   health_check_grace_period = 180
   launch_configuration      = aws_launch_configuration.api-lc.name
-  load_balancers            = [var.api_elb]
+  load_balancers            = var.api_elb_names
```
Comment (myazidi) [inline broker-deploy/broker-deploy.tf:21]: Did you test this out??
I think you can't have both `load_balancers` and `target_group_arns`, you will only need `target_group_arns`

File: broker-deploy/broker-deploy.tf [hunk 2]
```diff
@@ -58,7 +59,8 @@ resource "aws_autoscaling_group" "val-asg" {
   health_check_type         = "ELB"
   health_check_grace_period = 180
   launch_configuration      = aws_launch_configuration.val-lc.name
-  load_balancers            = [var.val_elb]
+  load_balancers            = var.val_elb_names
```
Comment (myazidi) [inline broker-deploy/broker-deploy.tf:62]: same as above

File: usaspending-deploy/usaspending-deploy.tf [hunk 3]
```diff
@@ -15,7 +15,8 @@ resource "aws_autoscaling_group" "api_asg" {
   health_check_type         = "ELB"
   health_check_grace_period = 30
   launch_configuration      = aws_launch_configuration.api_lc.name
-  load_balancers            = [var.api_elb]
+  load_balancers            = var.api_elb_names
```
Comment (myazidi) [inline usaspending-deploy/usaspending-deploy.tf:18]: same as [here](https://github.com/fedspendingtransparency/data-act-build-tools/pull/264/files#r468665899)

File: usaspending-deploy/usaspending-deploy.tf [hunk 4]
```diff
@@ -86,8 +87,9 @@ resource "aws_autoscaling_policy" "api_scale_up" {
   autoscaling_group_name = aws_autoscaling_group.api_asg.name
 }
 
-resource "aws_cloudwatch_metric_alarm" "api_alarm_high_requests" {
-  alarm_name          = "${var.api_name_prefix}_requestshigh (${var.aws_amis[var.aws_region]})"
+resource "aws_cloudwatch_metric_alarm" "api_alarm_high_requests_elb" {
+  count               = var.api_scaling_metric == "ELB" ? length(var.api_elb_names) : 0
```
Comment (myazidi) [inline usaspending-deploy/usaspending-deploy.tf:91]: Not sure if we are going to ever hit a case where the `api_elb_names` is more than one ??

File: usaspending-deploy/usaspending-deploy.tf [hunk 5]
```diff
@@ -135,6 +137,24 @@ resource "aws_cloudwatch_metric_alarm" "api_alarm_high_requests" {
   alarm_actions     = [aws_autoscaling_policy.api_scale_up.arn]
 }
 
+resource "aws_cloudwatch_metric_alarm" "api_alarm_high_requests_alb" {
+  count               = var.api_scaling_metric == "ALB" ? length(var.api_target_group_arns) : 0
```
Comment (myazidi) [inline usaspending-deploy/usaspending-deploy.tf:141]: same here, just one target group all the time

File: usaspending-deploy/usaspending-deploy.tf [hunk 6]
```diff
@@ -144,8 +164,9 @@ resource "aws_autoscaling_policy" "api_scale_down" {
   autoscaling_group_name = aws_autoscaling_group.api_asg.name
 }
 
-resource "aws_cloudwatch_metric_alarm" "api_alarm_low_requests" {
-  alarm_name          = "${var.api_name_prefix}_requestslow (${var.aws_amis[var.aws_region]})"
+resource "aws_cloudwatch_metric_alarm" "api_alarm_low_requests_elb" {
+  count               = var.api_scaling_metric == "ELB" ? length(var.api_elb_names) : 0
```
Comment (myazidi) [inline usaspending-deploy/usaspending-deploy.tf:168]: same as [here](https://github.com/fedspendingtransparency/data-act-build-tools/pull/264/files#r468670866)

File: usaspending-deploy/usaspending-deploy.tf [hunk 7]
```diff
@@ -193,6 +214,24 @@ resource "aws_cloudwatch_metric_alarm" "api_alarm_low_requests" {
   alarm_actions     = [aws_autoscaling_policy.api_scale_down.arn]
 }
 
+resource "aws_cloudwatch_metric_alarm" "api_alarm_low_requests_alb" {
+  count               = var.api_scaling_metric == "ALB" ? length(var.api_target_group_arns) : 0
```
Comment (myazidi) [inline usaspending-deploy/usaspending-deploy.tf:218]: same as [here](https://github.com/fedspendingtransparency/data-act-build-tools/pull/264/files#r468671837)

File: broker-deploy/broker-deploy.tf (hunk 1 above)
Comment (myazidi) [reply broker-deploy/broker-deploy.tf:21]: https://registry.terraform.io/providers/hashicorp/aws/latest/docs/resources/autoscaling_group#load_balancers

Comment (bminahan-kc) [reply broker-deploy/broker-deploy.tf:21]: you definitely can. Tested deployment in sandbox

File: usaspending-deploy/usaspending-deploy.tf (hunk 4 above)
Comment (bminahan-kc) [reply usaspending-deploy/usaspending-deploy.tf:91]: made it a list to future-proof for zero downtime load balancer migrations in the future. Ie:
- step 1: append new load balancer name to list
- step 2: change DNS name to new load balancer
- step 3: remove old load balancer form list

File: broker-deploy/broker-deploy.tf (hunk 1 above)
Comment (bminahan-kc) [reply broker-deploy/broker-deploy.tf:21]: https://ops.usaspending.gov:8080/job/usaspending-api-nonprod-deploy/1401/console

**Pull request merged**